crates/cast/bin/main.rs

```diff
@@ -489,11 +489,15 @@ async fn main() -> Result<()> {
             &mut std::io::stdout(),
         ),
         Subcommands::Logs(cmd) => cmd.run().await?,
-        Subcommands::Decode { tx } => {
+        Subcommands::DecodeTransaction { tx } => {
             let tx = stdin::unwrap_line(tx)?;
-            let (tx, sig) = SimpleCast::decode_tx(tx)?;
-            println!("Transaction: {}", tx);
-            println!("Signature: {}", sig);
+            let (tx, sig) = SimpleCast::decode_raw_transaction(&tx)?;
+
+            // Serialise tx and sig as json strings and reformat to print as a single json object
+            let mut tx = serde_json::to_string_pretty(&tx)?;
+            tx.truncate(tx.len() - 2);
+            let sig = serde_json::to_string_pretty(&sig)?[2..].to_string();
+            println!("{},\n{}", tx, sig);
```

Comment:
nit: both can be serialized into `to_value` and then merged like this:

```rust
fn merge(a: &mut Value, b: Value) {
    match (a, b) {
        (a @ &mut Value::Object(_), Value::Object(b)) => {
            let a = a.as_object_mut().unwrap();
            for (k, v) in b {
                merge(a.entry(k).or_insert(Value::Null), v);
            }
        }
        (a, b) => *a = b,
    }
}
```
https://stackoverflow.com/questions/47070876/how-can-i-merge-two-json-objects-with-rust

since we know the signature has no nested objects we don't need the recursive step

Reply:
super neat – thanks for pointing this out. will push a fix in just a bit

Reply:
just pushed, tried to refactor it to be more concise:
```rust
    // Serialize tx, sig and constructed a merged json string
    let mut tx = serde_json::to_value(&tx)?;
    let tx_map = tx.as_object_mut().unwrap();

    serde_json::to_value(sig)?.as_object().unwrap().iter().for_each(|(k, v)| {
        tx_map.entry(k).or_insert(v.clone());
    });
```